Source/MediaInfo/Image/File_Png.cpp

```diff
@@ -585,6 +590,34 @@ void File_Png::gAMA()
     Data_Common();
 }
 
+//---------------------------------------------------------------------------
+void File_Png::gdAT()
+{
+    Stream_Prepare(Stream_Image);
+    Fill(Stream_Image, StreamPos_Last, Image_Type, "Gain map");
+
+    //Parsing
+    File_Png MI;
+    Open_Buffer_Init(&MI);
+    Open_Buffer_Continue(&MI);
+    Open_Buffer_Finalize(&MI);
+    Merge(MI, Stream_Image, 0, StreamPos_Last);
+    GainMap_metadata_ISO.reset(new GainMap_metadata());
+    *static_cast<GainMap_metadata*>(GainMap_metadata_ISO.get()) = *static_cast<GainMap_metadata*>(MI.GainMap_metadata_ISO.get());

```

Comment:
```suggestion
    if (MI.GainMap_metadata_ISO.get()) {
        GainMap_metadata_ISO.reset(new GainMap_metadata());
        *static_cast<GainMap_metadata*>(GainMap_metadata_ISO.get()) = *static_cast<GainMap_metadata*>(MI.GainMap_metadata_ISO.get());
    }
```

else crash with https://skia.googlesource.com/skia/+/418c68ea5ccb/resources/images/gainmap_gdat_no_gmap.png